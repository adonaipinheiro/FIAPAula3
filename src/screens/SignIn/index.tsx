import React from 'react';
import {View, Text} from 'react-native';
import {Formik} from 'formik';
import {useTranslation} from 'react-i18next';

// Components
import {Input, Button} from '../../components';

// Styles
import {styles} from './styles';

// Hooks
import useSignIn from './hooks/useSignIn';

export default function SignIn() {
  const {initialValues, SignInSchema, submit, goToSignUp} = useSignIn();
  const {t} = useTranslation();

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{t('title')}</Text>
      <Text style={styles.subTitle}>{t('subTitle')}</Text>
      <Formik
        initialValues={initialValues}
        validationSchema={SignInSchema}
        onSubmit={(values, {setSubmitting}) => submit(values, setSubmitting)}>
        {({handleChange, handleSubmit, values, errors, isSubmitting}) => (
          <>
            <Input
              placeholder={t('signIn.inputEmail.text') || ''}
              value={values.email}
              onChangeText={handleChange('email')}
              error={errors.email}
              autoCapitalize="none"
              keyboardType="email-address"
            />
            <Input
              placeholder={t('signIn.inputPass.text') || ''}
              value={values.pass}
              secureTextEntry
              onChangeText={handleChange('pass')}
              error={errors.pass}
              autoCapitalize="none"
            />
            <Button
              type="secondary"
              disabled={isSubmitting}
              text={
                isSubmitting
                  ? t('signIn.loadingText')
                  : t('signIn.signInButton')
              }
              onPress={handleSubmit}
            />
            <Button
              type="primary"
              text={t('signIn.goToSignUpButton')}
              onPress={goToSignUp}
            />
          </>
        )}
      </Formik>
    </View>
  );
}
