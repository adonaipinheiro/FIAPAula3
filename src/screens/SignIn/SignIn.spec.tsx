import React from 'react';
import {render, screen} from '@testing-library/react-native';

import SignIn from '.';

jest.mock('@react-navigation/native');
jest.mock('react-redux');
jest.mock('react-i18next');
jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  },
}));

describe('SignIn', () => {
  beforeEach(() => {
    render(<SignIn />);
  });
  it('should render corretly', () => {
    const signInScreen = screen;
    expect(signInScreen).toBeTruthy();
  });
  it('should match snapshot', () => {
    const signInScreen = screen.toJSON();
    expect(signInScreen).toMatchSnapshot();
  });
});
