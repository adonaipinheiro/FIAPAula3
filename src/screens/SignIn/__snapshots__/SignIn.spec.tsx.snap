// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ed145b",
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#FFF",
        "fontFamily": "Roboto",
        "fontSize": 42,
      }
    }
  >
    title
  </Text>
  <Text
    style={
      {
        "color": "#FFF",
        "fontFamily": "Roboto",
        "fontSize": 18,
        "marginBottom": 30,
      }
    }
  >
    subTitle
  </Text>
  <View
    style={
      {
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      keyboardType="email-address"
      onChangeText={[Function]}
      placeholder="signIn.inputEmail.text"
      style={
        {
          "backgroundColor": "#FFFFFF90",
          "borderRadius": 8,
          "minHeight": 40,
          "padding": 10,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      {
        "marginTop": 20,
        "width": "100%",
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      onChangeText={[Function]}
      placeholder="signIn.inputPass.text"
      secureTextEntry={true}
      style={
        {
          "backgroundColor": "#FFFFFF90",
          "borderRadius": 8,
          "minHeight": 40,
          "padding": 10,
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFF",
        "borderRadius": 8,
        "justifyContent": "center",
        "marginTop": 20,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          {
            "color": "#be0e49",
          },
        ]
      }
    >
      signIn.signInButton
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#ed145b",
        "borderRadius": 8,
        "justifyContent": "center",
        "marginTop": 20,
        "minHeight": 40,
        "opacity": 1,
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Roboto",
            "fontSize": 16,
            "fontWeight": "bold",
          },
          {
            "color": "#FFF",
          },
        ]
      }
    >
      signIn.goToSignUpButton
    </Text>
  </View>
</View>
`;
